sydent/http/servlets/__init__.py

```diff
@@ -67,16 +77,17 @@ def get_args(request, required_args):
     # If we didn't get anything from that, and it's a v1 api path, try the request args
     # (riot-web's usage of the ed25519 sign servlet currently involves
     # sending the params in the query string with a json body of 'null')
-    if args is None and (v1_path or request.method == 'GET'):
-        args = copy.copy(request.args)
+    if args is None and (v1_path or request.method == b'GET'):
+        args_bytes = copy.copy(request.args)
         # Twisted supplies everything as an array because it's valid to
         # supply the same params multiple times with www-form-urlencoded
         # params. This make it incompatible with the json object though,
         # so we need to convert one of them. Since this is the
         # backwards-compat option, we convert this one.
-        for k, v in args.items():
+        args = {}
+        for k, v in args_bytes.items():
             if isinstance(v, list) and len(v) == 1:
-                args[k] = v[0]
+                args[k.decode("UTF-8")] = v[0].decode("UTF-8")
```

Comment:
it might be worth considering what happens when somebody provides a parameter that's not valid utf-8: I suspect it will make sydent return a 500 error, which isn't really appropriate. But this is a fairly minor issue, so :man_shrugging: